IanGRap/Squish
# Bugfix #106

Contributors: Austin

Goal
- Fix the bug where sometimes, when two players die in quick succession, the game displays Oozing Orange as the winner with a _white_ skelegoo icon (even if Oozing Orange is not in the game)

What Changed:
- In MatchManager.cs, set the winning player when elements are removed from the _alivePlayers array. This guarentees that we do not destroy the game object before trying to read from it.
- In UIMatchManager.cs, check if WinningPlayer != null to pull up game over screen, instead of checking if AlivePlayers.Count == 1

How to Test
- Test in development first
- Go to the New Lobby Scene and start a match
- Get to Game Point (someone has 2 points and will win if they win another match)
- Create this scenario (where the winning player is on the top)
    
![image](https://user-images.githubusercontent.com/28333213/54247819-a6f20300-44f7-11e9-91bc-67575767c9df.png)

- You should at this point see the error (if you are in development)
- Pull this branch, create the same scenario and see if the bug still pops up
